pkg/reconciler/apis/apibindingdeletion/apibinding_deletion_controller.go

```diff
@@ -78,10 +76,19 @@ func NewController(
 	queue := workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), controllerName)
 
 	c := &Controller{
-		queue:             queue,
-		metadataClient:    metadataClient,
-		kcpClusterClient:  kcpClusterClient,
-		apiBindingsLister: apiBindingInformer.Lister(),
+		queue: queue,
+		listCustomResources: func(ctx context.Context, cluster logicalcluster.Name, gvr schema.GroupVersionResource) (*metav1.PartialObjectMetadataList, error) {
+			return metadataClient.Resource(gvr).Namespace(metav1.NamespaceAll).List(genericapirequest.WithCluster(ctx, genericapirequest.Cluster{Name: cluster}), metav1.ListOptions{})
+		},
+		deleteCustomResources: func(ctx context.Context, cluster logicalcluster.Name, gvr schema.GroupVersionResource, namespace string) error {
+			background := metav1.DeletePropagationBackground
+			opts := metav1.DeleteOptions{PropagationPolicy: &background}
+			return metadataClient.Resource(gvr).Namespace(namespace).DeleteCollection(genericapirequest.WithCluster(ctx, genericapirequest.Cluster{Name: cluster}), opts, metav1.ListOptions{})
+		},
+		getAPIBinding: func(cluster logicalcluster.Name, name string) (*apisv1alpha1.APIBinding, error) {
+			return apiBindingInformer.Lister().Get(kcpcache.ToClusterAwareKey(cluster.String(), "", name))
+		},
+		commit: committer.NewCommitter[*APIBinding, *APIBindingSpec, *APIBindingStatus](kcpClusterClient.ApisV1alpha1().APIBindings()),
```

Comment:
Note for tomorrow: either have to undo this, or modify the controller so it modifies spec and status in 2 different passes